test/reports/test-report-manager.js

```diff
@@ -0,0 +1,290 @@
+const mockgoose = require('mockgoose');
+const mongoose = require('mongoose');
+const assert = require('assert');
+mockgoose(mongoose);
+
+const reportManager = require('../../src/reports/report-manager');
+const reportModels = require('../../src/reports/report-models');
+const Report = reportModels.Report;
+const EventReport = reportModels.EventReport;
+const OtherReport = reportModels.OtherReport;
+
+describe('reportManager', () => {
+  // Mockgoose doesn't start the mock until after 'mongoose.connect()' is
+  // called.
+  // TODO(jmtaber129): Find a version of mockgoose that works and doesn't
+  // print to console each time 'mongoose.connect()' is called.
+  before((done) => { mongoose.connect('', done); });
+
+  // Clear out the mocked database before each test case.
+  beforeEach((done) => {
+    mockgoose.reset();
+    done();
+  });
+
+  after((done) => { mongoose.unmock(done); });
+
+  describe('#createReport', () => {
+    it('should pass a created event report to the callback', (done) => {
+      const eventReport = {
+        type: 'event',
+        approvalStatus: true,
+        student: 'John Doe',
+        event: 'someEventUid',
+      };
+      
+      reportManager.createReport(eventReport, {}, (err, createdReport) => {
+        assert.equal(err, null);
+        assert.equal(createdReport.type, 'EventReport');
```

Comment:
```type``` is being used as the discriminator key, so that field is populated by mongoose as the name of the model.  Would it be better to change  the possible ```type``` fields of the objects passed to ```createReport()``` to the model names? (i.e., ```eventReport.type``` would be 'EventReport' instead of 'event', and ```createReport()``` would be changed accordingly).